### PR TITLE
Feature 112: Final-form for Boilers

### PR DESCRIPTION
**Pull request by ValentinSilvestri**
Final form finally added to Boilers.

### CODE REVIEW

File: src/components/Boilers/BoilersForm/index.js [hunk 1]
```diff
@@ -8,61 +10,115 @@ const BoilersForm = ({
...
+                    {meta.error && meta.touched && <div>{meta.error}</div>}
+                  </div>
+                )}
+              </Field>
+            </div>
+            <div className={styles.inputContainer}>
+              <label htmlFor="boilerType">Boiler Type:</label>
+              <Field name="boilerType" validate={required}>
+                {({ input, meta }) => (
+                  <div>
```
Comment (lmalvasia) [inline src/components/Boilers/BoilersForm/index.js:75]: You can use the shared component `SelectInput`

**Review by lmalvasia — CHANGES_REQUESTED** [review]

File: src/components/Boilers/BoilersForm/index.js (hunk 1 above)
Comment (ValentinSilvestri) [reply src/components/Boilers/BoilersForm/index.js]: Done.

File: src/components/Boilers/BoilersForm/index.js [hunk 2]
```diff
@@ -8,61 +10,115 @@ const BoilersForm = ({
...
+          hourEventualCost: boiler.hourEventualCost || 0,
+          maintenanceRate: boiler.maintenanceRate || 0,
+        }}
+        render={({ handleSubmit }) => (
+          <form className={styles.boilersFormContainer}>
+            <div className={styles.inputContainer}>
+              <label htmlFor="description">Description:</label>
+              <Field name="description" validate={required}>
+                {({ input, meta }) => (
+                  <div>
```
[the remark this answers is not in the document]
Comment (ValentinSilvestri) [reply src/components/Boilers/BoilersForm/index.js]: Done.

File: src/components/Boilers/BoilersForm/index.js [hunk 3]
```diff
@@ -8,61 +10,115 @@ const BoilersForm = ({
...
+      boilerType: values.boilerType.value,
+      hourMaintenanceCost: values.hourMaintenanceCost,
+      hourEventualCost: values.hourEventualCost,
+      maintenanceRate: values.maintenanceRate,
     };
-    onSubmit(boilerToSub, state.id);
+    onSubmit(boilerToSub, boiler._id);
   };
 
+  const required = (value) => (value ? undefined : 'Required');
```
[the remark this answers is not in the document]
Comment (ValentinSilvestri) [reply src/components/Boilers/BoilersForm/index.js]: Done.

File: src/components/Boilers/BoilersForm/index.js [hunk 4]
```diff
@@ -8,61 +10,115 @@ const BoilersForm = ({
...
 }) => {
-  const [state, setState] = useState({
-    description: boiler.description || '',
-    boilerType: boiler.boilerType || '5fcc1d06998cd913c71c7e01',
-    hourMaintenanceCost: boiler.hourMaintenanceCost || 0,
-    hourEventualCost: boiler.hourEventualCost || 0,
-    maintenanceRate: boiler.maintenanceRate || 0,
-    id: boiler._id,
-  });
+  const [boilerTypeOptions, setBoilerTypeOptions] = useState([]);
```
[the remark this answers is not in the document]
Comment (ValentinSilvestri) [reply src/components/Boilers/BoilersForm/index.js]: Done.

File: src/components/Boilers/BoilersForm/index.js [hunk 5]
```diff
@@ -8,61 +10,115 @@ const BoilersForm = ({
...
+  const [pickedBoilerType, setPickedBoilerType] = useState(null);
 
-  const onChangeInput = (e) => {
-    setState({
-      ...state,
-      [e.target.name]: e.target.value,
-    });
-  };
+  useEffect(async () => {
```
[the remark this answers is not in the document]
Comment (ValentinSilvestri) [reply src/components/Boilers/BoilersForm/index.js]: Partially done, got some issues with the select and the update.

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**